### PR TITLE
求職天眼通API 以公司搜尋

### PR DESCRIPTION
**Pull request by barry800414**
to solve #87 
- [x] code
- [x] test

API doc will be done in the future, please review it.

### CODE REVIEW

File: routes/clairvoyance/search.js [hunk 1]
```diff
@@ -1,14 +1,92 @@
...
+
+    const collection = req.db.collection('workings');
+
+    if (! company || company === '') {
+        next(new HttpError("company is required", 422));
+        return;
+    }
+
+    //mongodb query
+    var q = {
```
Comment (mark86092) [inline routes/clairvoyance/search.js]: Please use the es6 const here

File: routes/clairvoyance/search.js [hunk 2]
```diff
@@ -1,14 +1,92 @@
...
+        created_at: -1
+    };
+
+    //displayed fields
+    var opt = {
+        _id: 0,
+        job_title:1,
+        company: 1, 
+        created_at:1,
+        week_work_time: 1
```
Comment (mark86092) [inline routes/clairvoyance/search.js]: Two style to consistent

`job_title: 1` <-- space

`week_work_time: 1,` <-- "," every line end

File: routes/clairvoyance/search.js [hunk 3]
```diff
@@ -1,14 +1,92 @@
...
+        data.total_count = count;
+        data.total_page = Math.ceil(count / 25); 
+        return collection.find(q, opt).sort(s).skip(25 * page).limit(25).toArray();
+    }).then(function(workings) {
+        data.page = page;
+        data.workings = workings;
+
+        res.send(data);
+    }).catch(function(err) {
+        console.log(err);
```
Comment (mark86092) [inline routes/clairvoyance/search.js]: Oops, hide it :smile:

File: routes/clairvoyance/search.js [hunk 4]
```diff
@@ -1,14 +1,92 @@
...
+  "message": "keyword is required"
+}
+@return 500 (如果內部資料庫連線問題)
+{
+  "error": {},
+  "message": "Internal Server Error"
+}
+*/
 router.get('/by-company', function(req, res, next) {
-    next();
```
Comment (LittleWhiteYA) [inline routes/clairvoyance/search.js]: next() 不需要放在最後面嗎0.0?

File: routes/clairvoyance/search.js [hunk 5]
```diff
@@ -1,14 +1,91 @@
...
+{
+  "error": {},
+  "message": "keyword is required"
+}
+@return 500 (如果內部資料庫連線問題)
+{
+  "error": {},
+  "message": "Internal Server Error"
+}
+*/
```
Comment (mark86092) [inline routes/clairvoyance/search.js]: Can you remove these comments?

Comment (barry800414) [inline routes/clairvoyance/search.js]: So, is there any newer version? or I just remove them ?

Comment (mark86092) [inline routes/clairvoyance/search.js]: Yes, it based on your code, so I will update it

**Comment by mark86092** [comment]
Nice, thank you

**Pull request merged**